website/static/js/fangorn.js

```diff
@@ -833,6 +836,45 @@ function expandStateLoad(item) {
     reapplyTooltips();
 }
 
+/**
+ * @param tree A Treebeard _item object for the row
+ * @param nodeID Current node._id
+ * @param file window.contextVars.file object
+ */
+function findCurrentFileID(tree, nodeID, file) {
```

Comment:
This name is a bit of a misnomer:  `find*` implies that this is doing some sort of search and returning an ID, but it is not. It has a number of side effects, updating `folderIndex`, updating folders, etc.